modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryArray.java

```diff
@@ -251,6 +253,17 @@ public String componentClassName() {
             && Arrays.deepEquals(this.arr, arr.arr);
     }
 
+    /** {@inheritDoc} */
+    @Override public int compareTo(@NotNull BinaryArray o) {
+        if (componentTypeId() != o.componentTypeId()) {
```

Comment:
Can `componentTypeId()` return changing value from `ctx.typeId(compClsName)` after component registration? "This can happen when binary type was not registered in time of binary array creation." Can we meet different comparisons on the same binary arrays?

Reply:
> Can componentTypeId() return changing value from ctx.typeId(compClsName) after component registration?

No.

Mapping `compClsName -> typeId` should be idempotent and equal on all nodes, by contract. 

> Can we meet different comparisons on the same binary arrays?

Logic with "use typeId or typeId(compClsName)" implemented to overcome this case.